### PR TITLE
Update Readme

### PR DESCRIPTION
**Pull request by nprizal**
bktec has grown into 4 test runners which own default configuration values and available features. This PR restructure the README by introducing the usage guide for each test runner. I choose a feature based guide as opposed to environment variable based guide, to make it easier to see how to configure a feature rather than having to look at the list of environment variables.

I have also removed the "Migrating to 1.0.0" section and replaced it with a link to the original content. v1.0.0 has been released for a while, so I don't think we need to put this information up front any more.

### CODE REVIEW

**Review by niceking — COMMENTED** [review]
I like the restructure, good call! 

Not much blocking - mostly a comment around the include/exclude patterns section which I think could be clarified and be written for a non-bk audience

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,7 +16,7 @@
 ## 0.9.0 - 2024-09-11
 - ⚠️ **BREAKING** Add the `BUILDKITE_SPLITTER_RESULT_PATH` required environment variable. See [Migrating to 0.9.0](https://github.com/buildkite/test-splitter/tree/db4cab8cd6c82392553cd80481cf75e3888c2f4c?tab=readme-ov-file#migrating-to-090).
 - Experimental support for Jest by setting `BUILDKITE_SPLITTER_TEST_RUNNER` to `jest`.
-- Update the retry behavior to only retry failed tests.
+- Update the retry behavior to only Automatically retry failed tests.
```
[the remark this answers is not in the document]
Comment (nprizal) [reply CHANGELOG.md:19]: Oh sorry, this not intentional. I will remove.

File: README.md [hunk 2]
```diff
@@ -1,37 +1,18 @@
...
-
-  In addition, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_TEST_ENGINE_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command. 
+| Feature                                            | Rspec | Jest | Playwright | Cypress |
+| -------------------------------------------------- | :---: | :--: | :--------: | :-----: |
+| Discover and filter test files using glob pattern  |   ✅  |   ✅  |    ✅      |    ✅   |
+| Automatically retry failed test                    |   ✅  |   ✅  |    ✅      |    ❌   |
+| Split slow files by individual test example        |   ✅  |   ❌  |    ❌      |    ❌   |
 
-Furthermore, version 1.0.0 introduces name changes to environment variables. To migrate to v1.0.0, You will need to update the following environment variables in your Pipeline:
+## Migrating to 1.0.0
```
[the remark this answers is not in the document]
Comment (nprizal) [reply README.md:13]: Sounds good to me. We have a link to this content from the CHANGELOG as well. I agree that we can remove this from README

File: README.md [hunk 3]
```diff
@@ -1,37 +1,18 @@
...
-
-- `BUILDKITE_TEST_ENGINE_RESULT_PATH`
-
-  The location of where the runner should store test results. We introduced a new feature that requires bktec to read test results from the runner for retries and verification purposes. To enable this feature, it is necessary to configure the `BUILDKITE_TEST_ENGINE_RESULT_PATH` environment variable.
-
-  In addition, we have updated the default test command for RSpec to `bundle exec rspec --format progress --format json --out {{resultPath}} {{testExamples}}`. Test Splitter will automatically replace `{{resultPath}}` with the value specified in `BUILDKITE_TEST_ENGINE_RESULT_PATH`. If you want to customize the RSpec command, make sure to include `--format json --out {{resultPath}}` in the command. 
+| Feature                                            | Rspec | Jest | Playwright | Cypress |
+| -------------------------------------------------- | :---: | :--: | :--------: | :-----: |
+| Discover and filter test files using glob pattern  |   ✅  |   ✅  |    ✅      |    ✅   |
```
[the remark this answers is not in the document]
Comment (nprizal) [reply README.md:9]: That his sounds better! Discover and filter is not very clear.

File: docs/cypress.md [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+## Discover and filter test files
+bktec discovers the test files using a glob pattern. By default, it identifies the files matching the `**/*.cy.{js,jsx,ts,tsx}` pattern. This means it will recursively find all JavaScript or TypeScript files with a `.cy` suffix, such as `/cypress/e2e/login.cy.ts`. You can customize this pattern using the `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN` environment variable.
+
+Additionally, you can exclude certain files or directories that match a specific pattern using the `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN` environment variable.
+
+To customize the discovery pattern and exclude certain files, set the following environment variable:
+```sh
+export BUILDKITE_TEST_ENGINE_TEST_PATTERN=**/*.cy.{ts,tsx}
+export BUILDKITE_TEST_ENGINE_TEST_EXCLUDE_PATTERN=cypress/component
```
[the remark this answers is not in the document]
Comment (nprizal) [reply docs/cypress.md:30]: That's right. They don't have to be used together. You can use `..TEST_FILE_PATTERN` or `..TEST_FILE_EXCLUDE_PATTERN` or both. I'll update this part.

[the remark this answers is not in the document]
Comment (nprizal) [reply docs/cypress.md:30]: Before I update for all runners, what do you think of the following changes?

---


## Filter test files
By default, bktec runs test files that match the `spec/**/*_spec.rb` pattern. You can customize this pattern using the `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN` environment variable. For instance, to configure bktec to only run test files inside the `spec/features` directory, use:

```sh
export BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN=spec/features/**/*_spec.rb
```

Additionally, you can exclude specific files or directories that match a certain pattern using the `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN` environment variable. For example, to exclude test files inside the `spec/features` directory, use:

```sh
export BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN=spec/features
```

You can also use both `BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN` and `BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN` simultaneously. For example, to run all test files inside the `spec/models` directory except those inside `spec/models/user`, use:

```sh
export BUILDKITE_TEST_ENGINE_TEST_FILE_PATTERN=spec/models/**/*_spec.rb
export BUILDKITE_TEST_ENGINE_TEST_FILE_EXCLUDE_PATTERN=spec/models/user
```

[the remark this answers is not in the document]
Comment (niceking) [reply docs/cypress.md:30]: Nice, that's way clearer ✨ Good to give the example of doing both as well!!!!!!!

**Review by niceking — APPROVED** [review]

**Pull request merged**